src/Simple.OData.Client.Core/Expressions/FunctionToOperatorMapping.cs

```diff
@@ -48,6 +50,8 @@ public override string Format(ExpressionContext context, ODataExpression functio
             return $"{functionArguments[0].Format(context)} in {listAsString}";
         }
 
+        internal override int Precedence => 2;
```

Comment:
To ensure I understand this code: now all functions will have a precedence of 2? (same as multiply, modulo, and divide)

I tried reading the OData spec to get a feel for where it sits in the precedence: http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part2-url-conventions.html#_Toc31361036

It appears instead this should have a precedence of 1 and all the existing ones be shifted down to 2, 3, ..., etc.

Reply:
I've added https://github.com/OData/odata.net/issues/2016 to discuss this oddity.

Reply:
Yes you are right, it seems to be an issue in OData/odata.net. Because of higher precedence of 'in' no braces should be necessary. I think we could use a workaround like `foo.Filter(x => !(true && x.Status.Contains("NotSoGood", "NotEvenBetter")))` that produces a query `?$filter=not (true and Status in ('NotSoGood', 'NotEvenBetter'))`, this query raises no errors and returns expected results.
Should I close this pull request?

Reply:
Clearly it is inconvenient that OData.Net does not support this, and it certainly makes life harder for our users. It may be reasonable for our code to introduce extraneous parentheses simply to disambiguate this situation.

Perhaps NeedsGrouping could always return true for nested `in` and `has` expressions?

Reply:
I have a PR against OData.net to resolve this: https://github.com/OData/odata.net/pull/2017

Reply:
I have reverted the changes in `ODataExpression.Format` and `FunctionToOperatorMapping` and added grouping per default to the 'in' operator as workaround.